addons/web_editor/static/src/js/summernote.js

```diff
@@ -1872,23 +1872,46 @@ $.summernote.pluginEvents.removeFormat = function (event, editor, layoutInfo, va
 var fn_boutton_updateRecentColor = eventHandler.modules.toolbar.button.updateRecentColor;
 eventHandler.modules.toolbar.button.updateRecentColor = function (elBtn, sEvent, sValue) {
     fn_boutton_updateRecentColor.call(this, elBtn, sEvent, sValue);
-    var font = $(elBtn).closest('.note-color').find('.note-recent-color i')[0];
-
-    if (sEvent === "foreColor") {
-        if (sValue.indexOf('text-') !== -1) {
-            font.className += ' ' + sValue;
-            font.style.color = '';
+    var recentcolor = $.find('.note-recent-color i');
+    var recentcolorbtn = $.find('.note-recent-color');
+
+    for (var i in recentcolor) {
+        var font = recentcolor[i];
+        var $button = $(recentcolorbtn[i]);
+        var className = font.className.split(/\s+/);
+        if (sEvent === "foreColor") {
+            for (var k=2; k<className.length; k++) {
+                if (className[k].length && className[k].slice(0,5) === "text-") {
+                  className.splice(k,1);
+                  k--;
+                }
+            }
+            if (sValue.indexOf('text-') !== -1) {
+                font.className = className.join(' ') + ' ' + sValue;
+                font.style.color = '';
+            } else {
+                font.className = font.className.replace(/(^|\s+)text-\S+/, '');
+                font.style.color = sValue !== 'inherit' ? sValue : "";
+            }
         } else {
-            font.className = font.className.replace(/(^|\s+)text-\S+/);
-            font.style.color = sValue !== 'inherit' ? sValue : "";
+            for (var k=2; k<className.length; k++) {
+                if (className[k].length && className[k].slice(0,3) === "bg-") {
+                  className.splice(k,1);
+                  k--;
+                }
+            }
+            if (sValue.indexOf('bg-') !== -1) {
+                font.className =className.join(' ') + ' ' + sValue;
+                font.style.backgroundColor = "";
+            } else {
+                font.className = font.className.replace(/(^|\s+)bg-\S+/, '');
+                font.style.backgroundColor = sValue !== 'inherit' ? sValue : "";
+            }
         }
-    } else {
-        if (sValue.indexOf('bg-') !== -1) {
-            font.className += ' ' + sValue;
-            font.style.backgroundColor = "";
-        } else {
-            font.className = font.className.replace(/(^|\s+)bg-\S+/, '');
-            font.style.backgroundColor = sValue !== 'inherit' ? sValue : "";
+        if(sValue !== 'inherit'){
+            var colorInfo = JSON.parse($button.attr('data-value'));
+            colorInfo[sEvent] = sValue;
+            $button.attr('data-value', JSON.stringify(colorInfo));
```

Comment:
This code is to set recentcolor button according to last used color for font or icon. both icon and font must suggest or set the same color in recentcolor button.